### ISSUE TITLE
PF11 cluster iptables don't start

### ISSUE THREAD

**Issue by olmanolsztyn** [inverse-inc/packetfence#6582]
**Describe the bug**
I deployed fresh pf11 cluster installation and packetfence-iptables don't start

**To Reproduce**
1. systemctl restart packetfence-iptables
2. systemctl status packetfence-iptables
3. See error: -e(13007) FATAL: Can't locate object method "STORE" via package "pfconfig::cached_array" at /usr/local/pf/lib/pf> (pf::iptables::generate_filter_if_src_to_chain) Can't locate object method "STORE" via package "pfconfig::cached_array" at /usr/local/pf/lib/pf/iptables.pm line 327."
4. /usr/local/pf/bin/pfcmd service iptables status
5. See packetfence-iptables.service                            stopped   0

**Comment by julsemaan** [comment]
We need more information on how to replicate the issue since that isn't happening when I test it in lab.

Please provide:
1. Exact package version you're running `dpkg -l | grep packetfence`
2. pf.conf
3. networks.conf
4. cluster.conf

Also please provide which step of your cluster build you're at

**Comment by nqb** [comment]
@julsemaan, now you assume everybody is running Debian :joy:
My 2 cents, I deployed last week a Debian cluster running PF 11 using minimal Debian net install and I don't got any issues with iptables.

**Comment by olmanolsztyn** [comment]
**1.**
dpkg -l | grep packetfence
- packetfence 11.0.0+20210915122246+371752348+0011+maintenance~11.0+bullseye1 all          PacketFence network registration / worm mitigation system
- packetfence-captive-portal-javascript 11.0.0+20210915122246+371752348+0011+maintenance~11.0+bullseye1 amd64        PacketFence captive portal javascript
- packetfence-config 11.0.0+20210915122246+371752348+0011+maintenance~11.0+bullseye1 all          Service use to manage PacketFence configuration.
- packetfence-doc 11.0.0+20210915122246+371752348+0011+maintenance~11.0+bullseye1 all          documentation for packetfence
- packetfence-golang-daemon 11.0.0+20210915122246+371752348+0011+maintenance~11.0+bullseye1 amd64        PacketFence Golang binary.
- packetfence-ntlm-wrapper 11.0.0+20210915122246+371752348+0011+maintenance~11.0+bullseye1 amd64        C wrapper around the ntlm_auth utility to log authentication latency and success/failure.
- packetfence-perl 1.1.2  all packetfence-perl
- packetfence-pfappserver-javascript 11.0.0+20210915122246+371752348+0011+maintenance~11.0+bullseye1 amd64        PacketFence Admin javascript css
- packetfence-pfcmd-suid 11.0.0+20210915122246+371752348+0011+maintenance~11.0+bullseye1 amd64 C wrapper that replace perl-suid dependence
- packetfence-redis-cache 11.0.0+20210915122246+371752348+0011+maintenance~11.0+bullseye1 all Init script to manage redis server.

**2.**
/usr/local/pf/conf/pf.conf
[general]
domain=xxx.xxx.xx
hostname=sso
timezone=Europe/Warsaw

[database]
host=127.0.0.1
pass=xxxxxxx

[services]
pfdhcp=disabled

[advanced]
configurator= <<EOT
disabled
EOT
openid_attributes=

[webservices]
user=packet
pass=xxxxxxx

[active_active]
password=xxxxxxx
galera_replication_username=pfcluster
galera_replication_password=xxxxxxx

[interface eth0.229]
ip=172.22.30.41
mask=255.255.255.0
type=management,high-availability

[interface eth0]
type=other,high-availability
mask=255.255.255.0
**3.** /usr/local/pf/conf/networks.conf is empty because i don't have any at this moment.
**4.** [CLUSTER]

management_ip=172.22.30.40

[CLUSTER interface eth0]

[CLUSTER interface eth0.229]
ip=172.22.30.40

[PF-PRIMARY]
management_ip=172.22.30.41

[PF-PRIMARY interface eth0]

[PF-PRIMARY interface eth0.229]
ip=172.22.30.41

[PF-SECONDARY]
management_ip=172.22.30.42

[PF-SECONDARY interface eth0]

[PF-SECONDARY interface eth0.229]
ip=172.22.30.42

**Comment by olmanolsztyn** [comment]
In addition i used packetfence ZEN installation.

**Comment by julsemaan** [comment]
I think the issue is that high-availability is defined on `eth0` but it doesn't have any IP address

Can you try removing it from there and do:
```
/usr/local/pf/bin/pfcmd pfconfig clear_backend
systemctl restart packetfence-config
/usr/local/pf/bin/pfcmd configreload hard
systemctl restart packetfence-iptables
```

Lets see if that fixes it, if so, we'll have to add proper handling of this case so that the HA interface is skipped in iptables when there is no IP on it

**Comment by olmanolsztyn** [comment]
Yep that was it. There should be fix that will prevent from this.

**Comment by julsemaan** [comment]
I'll replicate it on my machine and push a fix to devel+maintenance

**Comment by julsemaan** [comment]
It is be fixed in maintenance/11.0, the new package should be available tomorrow assuming the pipeline succeeds tonight

**Comment by olmanolsztyn** [comment]
I have one suggestion about gui to gray out ha option in interface if there is no ip address.
